app/ante/ante.go

```diff
@@ -46,6 +46,8 @@ func NewAnteHandler(options HandlerOptions) (sdk.AnteHandler, error) {
 		return nil, err
 	}
 
+	ethAnteHandler := newEthAnteHandler(options)
```

Comment:
can move inside the case?

Reply:
It's moved here so the result is cached

Reply:
although after the refactoring, there's not much difference anymore, since it's simply a closure now, but previously there are a bunch of decrator objects.